code/game/objects/items/robot/robot_items.dm

```diff
@@ -146,8 +146,65 @@
 
 		user << "<span class='notice'>You stop charging [target].</span>"
 
+/obj/item/device/harmalarm
+	name = "Sonic Harm Prevention Tool"
+	desc = "Releases a harmless blast that confuses most organics. For when the harm is JUST TOO MUCH"
+	icon_state = "megaphone"
+	var/cooldown = 0
+	var/emagged = 0
+
+/obj/item/device/harmalarm/emag_act(mob/user)
+	emagged = !emagged
+	if(emagged)
+		user << "<font color='red'>You short out the safeties on the [src]!</font>"
+	else
+		user << "<font color='red'>You reset the safeties on the [src]!</font>"
+
+/obj/item/device/harmalarm/attack_self(mob/user)
+	var/safety = !emagged
+	if(cooldown > world.time)
+		user << "<font color='red'>The device is still recharging!</font>"
+		return
 
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.cell.charge < 1200)
+			user << "<font color='red'>You don't have enough charge to do this!</font>"
+			return
+		R.cell.charge -= 1000
+		if(R.emagged)
+			safety = 0
+
+	if(safety == 1)
+		for(var/mob/living/M in get_hearers_in_view(9, user))
+			if(iscarbon(M))
+				M.confused += 6
+				M << "<font color='red' size='3'>MY EARS</font>"

```

Comment:
use visible message instead of your custom speak loop

plus use the standard spans instead of encoding the styles